alphagov/paas-elasticache-broker
# [#183045327] Allow user to specify a maintenance window

What
-----

Allow users to specify a maintenance window when creating or updating a redis service

How to review
-------------

* Code Review
* Deploy to a dev environment (Currently deployed to dev02)
    * **Create with a specific window**
        * Provision a service with a specified maintenance window: `-c '{"preferred_maintenance_window": "mon:11:11-wed:10:10"}'` (or some similar time)
        * Ensure that the service is created with this maintenance window with cloudfoundry: `cf service $SERVICE_NAME --params`
        * Ensure that the service is created with this maintenance window using the AWS console
    * **Update to new window**
       * Update the service, specifying a new window: `-c '{"preferred_maintenance_window": "fri:11:11-fri:13:13"}'` (or some similar time)
        * Ensure that the service is updated with this maintenance window with cloudfoundry: `cf service $SERVICE_NAME --params`
        * Ensure that the service is updated with this maintenance window using the AWS console
